### ISSUE TITLE
nodeFunctions created in other packages can't find functions from nimble namespace

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#894]
See bottom for an example.

The enclosing env for nodeFunctions defaults to globalenv() based on the where argument.

A few options for handling this (where #3 is probably the correct approach but we may want to have Daniel use #1 for now):

1) Have packages use Depends:nimble. 

2) Embedding nimble:: in generated code would be some work and would then involve stripping it out when compiling.

3) A final idea which one of you already anticipated is that `buildNodeFunctions` has a `where` argument that is defaulting to `globalenv`.  We could rework the generation of all nodeFunctions have `where = getNimbleFunctionEnvironment()`. E.g. this would then be BayesNSGP and BayesNSGP's parent is `imports:nimble`.

```
> Rmodel <- nsgpModel(likelihood = "fullGP", constants = constants, coords = coords, z = z )
defining model...
Adding ones, dist1_sq, dist2_sq, dist12 as data for building model.
building model...
setting data and initial values...
running calculate on model (any error reports that follow may simply reflect missing values in model variables) ... Error in dmnorm_chol(model$z[1:100], mean = model$mu[1:100], cholesky = model$lifted_chol_oPC_oB1to100_comma_1to100_cB_cP[1:100,  : 
  could not find function "dmnorm_chol"

checking model sizes and dimensions...Error in nimSwitch(paramID = PARAMID_, IDoptions = 1:5, value = PARAMANSWER_ <- model$alpha[1],  : 
  could not find function "nimSwitch"
Error in nimSwitch(paramID = PARAMID_, IDoptions = 1:5, value = PARAMANSWER_ <- model$alpha[1],  : 
  could not find function "nimSwitch"
```

**Comment by perrydv** [comment]
Thanks for tracking this down @paciorek .  Option 3 makes sense to me.  It looks like the `where` arguments are there (from `model$buildNodeFunctions` to `nodeFunctionNew` to `nimbleFunction`) and just aren't ending up with a useful default for the case of appearing in other package code.

**Comment by paciorek** [comment]
Ok, let's plan on that, but it may not be wise/realistic to do this for this release, particularly given we'd be monkeying with something at the core of nimble's functionality. 

I'll propose that @danielturek or I could experiment with this in the next couple weeks on a separate branch.

**Comment by paciorek** [comment]
I think I dealt with this. Calls to `BayesNSGP::nsgpModel` are fine. Strange thing is that this issue was filed after PR #889  resolved issue #876.

@danielturek  let me know if you have any better memory here or any reason I shouldn't close this.

**Comment by danielturek** [comment]
I believe it was resolved also.  I'm not sure about the ordering of the events, though..
